platform/admin/admin-configuration-configupdater/src/main/java/org/codice/ddf/admin/configuration/ConfigurationUpdater.java

```diff
@@ -0,0 +1,333 @@
+                doHandleStore(context);
+                return null;
+              });
+    } catch (PrivilegedActionException e) {
+      // Only caught if the action through a CHECKED exception, thus it MUST be an IOException.
+      throw new IOException(e);
+    }
+  }
+
+  /**
+   * Sync the config files in etc when an update occurs. There are four cases, which will be denoted
+   * using 2-tuple combinations of variables: [felixFileFromCache, felixFileFromConfig]
+   *
+   * <p>[null, null] (1) Config was created from config admin directly; not by dropping a file.
+   * There is nothing to track.
+   *
+   * <p>[null, X] (2) Some config was just created by dropping a file into etc; track the file.
+   * Update the properties on disk if necessary and perform any special first time processing.
+   *
+   * <p>[Y, null] [Y, X] (3) Felix prop was externally changed or removed. This is an error.
+   *
+   * <p>[Y, Y] (4) Felix file prop did not change, do nothing. This is the happy path. Properties
+   * will be updated on disk if necessary.
+   */
+  private void doHandleStore(ConfigurationContext context) throws IOException {
+    String pid = context.getServicePid();
+    File fileFromConfig = context.getConfigFile();
+
+    CachedConfigData cachedConfigData = pidDataMap.get(pid);
+    File fileFromCache = (cachedConfigData != null) ? cachedConfigData.getFelixFile() : null;
+
+    if (fileFromCache == null && fileFromConfig == null) {
+      // This config doesn't have an etc file, so we ignore this case (1)
+      LOGGER.debug("Was not tracked and will not track pid {}", pid);
+      return;
+    }
+
+    if (fileFromCache == null) {
+      // An etc config file was just dropped and we're seeing it for the first time (2)
+      LOGGER.debug("Tracking pid {}", pid);
+      CachedConfigData createdConfigData = new CachedConfigData(context);
+      pidDataMap.put(pid, createdConfigData);
+      writeIfNecessary(
+          context.getServicePid(),
+          fileFromConfig,
+          context.getSanitizedProperties(),
+          createdConfigData);
+      return;
+    }
+
+    guardAgainstFelixFilePropChanging(fileFromConfig, fileFromCache);
+
+    // Routine property updates for tracked files - write to disk if necessary (4)
+    writeIfNecessary(
+        context.getServicePid(),
+        fileFromConfig,
+        context.getSanitizedProperties(),
+        cachedConfigData);
+  }
+
+  private void guardAgainstFelixFilePropChanging(File fileFromConfig, File fileFromCache) {
+    if (!Objects.equals(fileFromConfig, fileFromCache)) {
+      // The felix file prop changed, which is not allowed (3)
+      String msg =
+          (fileFromConfig == null)
+              ? format("Felix filename has been illegally removed, was [%s]", fileFromCache)
+              : format(
+                  "Felix filename has been illegally changed from [%s] to [%s]",
+                  fileFromCache, fileFromConfig);
+      throw new IllegalStateException(msg);
```

Comment:
❗️ Have you run itests against this change? I remember that when attempting etc sync last time, itests failed because they were touching this case. And we decided that, if this happens, we update the in-memory config with the old felix.fileinstall property and leave the property alone on disk.

Reply:
Added a note to verify itests. Previously they failed on init due to duplication, so I provided an override. It was only because of Scott's old password encryption module. 

Fairly certain itests do not hijack the felix file name since we don't ever do that in production.

Reply:
They aren't doing anything with the felix file name specifically. But I believe they are updating in-memory configs with the normal properties but leaving off the "system properties". Which leads to the config's felix file name being removed and thus getting to this case.